examples/jsm/postprocessing/UnrealBloomPass.js

```diff
@@ -190,7 +190,7 @@ UnrealBloomPass.prototype = Object.assign( Object.create( Pass.prototype ), {
 			this.renderTargetsHorizontal[ i ].setSize( resx, resy );
 			this.renderTargetsVertical[ i ].setSize( resx, resy );
 
-			this.separableBlurMaterials[ i ].uniforms[ "texSize" ].value = new Vector2( resx, resy );
+			this.separableBlurMaterials[ i ].uniforms[ "texSize" ].value = new Vector2( resx / window.devicePixelRatio, resy / window.devicePixelRatio );
```

Comment:
There's no guarantee of what a user has set the pixel ratio of the effect composer to or that they have set it at all in which case the effect will be inconsistent again so this doesn't feel like an adequate fix. I often change the pixel ratio dynamically, as well, depending on the run time performance of my application.

The [postprocessing package](https://github.com/vanruesc/postprocessing) deals with this by using a fixed render target resolution so the effect is consistent relative to the screen size. You can see an example of it in the [blur demo](https://vanruesc.github.io/postprocessing/public/demo/#blur). Something similar may be a better solution here.

Reply:
> I often change the pixel ratio dynamically, as well, depending on the run time performance of my application.

Hello @gkjohnson, thank you for your information, I'll research it. But why you change window.devicePixelRatio? Although it can be changed, but I think this parameter should be a parameter of a physical device and should not be changed. If really needed, can other new variable be used instead?

Reply:
I hink he meant he changes the value on the composer, so you cant assume that it equals to window.devicePixelRatio that you used here

Reply:
@makc OK, thanks! I'll research more.

Reply:
> I think he meant he changes the value on the composer, so you cant assume that it equals to window.devicePixelRatio that you used here

That's right -- I adjust it based on performance because it can have a significant impact. On some devices the pixel ratio is set to 3 or 4 so lowering it can help a lot without really sacrificing visuals.

Reply:
@gkjohnson Thanks! I directly use the Resizer.js of [vanruesc/postprocessing repository](https://github.com/vanruesc/postprocessing/blob/master/src/core/Resizer.js). It works very well, but I’m not sure if this is allowed.
https://raw.githack.com/gonnavis/three.js/40b1da13cfeb52fab0cbec0dd6544f6121f354d2/examples/webgl_postprocessing_unreal_bloom.html

Reply:
Sorry but I think this goes in the wrong direction. The user is supposed to manage `UnrealBloomPass.resolution` on application level.

Reply:
From my experience, I worked a lot to find out the problem after adjusting various parameters such as exposure, strength and radius.

Reply:
@Mugen87 Oh yes, it is not possible to change bloomPass.x at the application level now. The code of Resizer.js needs to be modified to maintain the naming consistency of width, x, height, y.

Reply:
> I worked a lot to find out the problem

I think they all like that part, it was great investigation work :D they just did not like the fix 🤷

Reply:
@Mugen87 I found that, at least for this UnrealBloomPass, `resolution` is only used for initialization, and the initialized value will be overwritten by `setSize()` immediately. Because when` composer.addPass( bloomPass );` is called, composer will automatically call `setSize()`.

Reply:
I added a `this.resizer`, keep `this.resolution` untouched.

Reply:
I think that the initialization code in the constructor should also be deleted, because they will be overwritten by the code in `setSize()` immediately, there is no need to exist, and the code complexity can be simplified.

Reply:
I found that this Resizer.js solution is fixed height, which will be very wasteful of performance under small size and wide-screen-ratio.

![image](https://user-images.githubusercontent.com/10785634/102057890-9fc4fb00-3e29-11eb-8ad7-2c120bb5c0a6.png)

Reply:
I think using Resizer.js will introduce too much complexity, and it also has performance problems under low resolution, so I don't want to use Resizer.js.

How about let `EffectCompoer` pass `pixelRatio` to `Pass`s through `setSize()`?
Thus, `Pass` can aware of the current `pixelRatio`, even if it is manually modified?

https://github.com/mrdoob/three.js/pull/20798/files/b13d25e07bb466b22c96c808b5ce0c72a5d37ef7